include/boost/geometry/arithmetic/arithmetic.hpp

```diff
@@ -51,17 +52,24 @@ struct value_operation
         : m_value(value)
     {}
 
-    template <typename P, int I>
-    inline void apply(P& point) const
+    template <typename PointDst, int I>
+    inline void apply(PointDst& dest_point) const
     {
```

Comment:
Abbreviations used (Dst, dest) are not the same.
While you are at it, you might change I to Index and make type std::size_t

What is C ?